pokemon-checker/src/services/PokemonService.ts

```diff
@@ -93,7 +93,9 @@ export class PokemonService {
     } else {
       const data = await response.json();
       const pokemonPayload = data as IPokemonData;
-      return this.factory.createPokemon(pokemonPayload);
+      const pokemon = this.factory.createPokemon(pokemonPayload);
+      await this.factory.fetchAbilities(pokemon);
```

Comment:
I feel like if the service wants full ability data, it should be the factory's job to do that setup.

The service tells the factory to "create a pokemon", and if one of the requirements for that means "have all ability data fetched/ready", there is work to be done by the factory.

That way, when the service says `this.factory.createPokemon`, it knows what it's getting is a finished product.